webrender/src/renderer.rs

```diff
@@ -2532,6 +2528,8 @@ impl<B: hal::Backend> Renderer<B> {
             samplers
         };
 
+        let mut frame_semaphore = self.device.device.create_semaphore();
+        let _frame = self.device.set_next_frame_id_and_return_frame(&mut frame_semaphore);
```

Comment:
You can create the semaphore in this function and only return that. (I think :) )